src/lib/Zikula/Core/AbstractTheme.php

```diff
@@ -77,6 +77,51 @@ public function generateThemedBlock(array $block)
     }
 
     /**
+     * load the themevars into the themeEngine global vars
+     */
+    public function setThemeVars()
+    {
+        $this->getContainer()->get('zikula_core.common.theme.themevars')->replace($this->getThemeVars());
+    }
+
+    /**
+     * Get the theme variables from both the DB and the .yml file.
+     * @return array|string
+     */
+    public function getThemeVars()
+    {
+        $dbVars = \ModUtil::getVar($this->name);
+        $defaultVars = $this->getDefaultThemeVars();
+        $combinedVars = array_merge($defaultVars, $dbVars);
+        if (array_keys($dbVars) != array_keys($combinedVars)) {
+            // First load of file or vars have been added to the .yml file.
+            \ModUtil::setVars($this->name, $combinedVars);
+        }
+
+        return $combinedVars;
+    }
+
+    /**
+     * Get the default values from variables.yml.
+     * @return array
+     */
+    public function getDefaultThemeVars()
+    {
+        $defaultVars = [];
+        $themeVarsPath = $this->getConfigPath() . '/variables.yml';
+        if (file_exists($themeVarsPath)) {
+            if ($this->getContainer()) {
+                $yamlVars = Yaml::parse($themeVarsPath);
```

Comment:
This shold be changed to ` Yaml::parse(file_get_contents('/path/to/file.yml'));`
> Because it is currently possible to pass a filename to this method, you must validate the input first. Passing a filename is deprecated in Symfony 2.2, and will be removed in Symfony 3.0.
-- http://symfony.com/doc/current/components/yaml/introduction.html